### ISSUE TITLE
Geolocator.getLastKnownPosition always returns [0.0, 10.69]

### ISSUE THREAD

**Issue by erf** [Baseflow/flutter-geolocator#962]
## 🐛 Bug Report

`Geolocator.getLastKnownPosition` always returns [0.0, 10.69], lat, lng pos.

This is a weird location and i've never been there.

Event after i get a position using `getCurrentPosition`, i still get that same weird location on `getLastKnownPosition`.

I'm on iOS.

### Expected behavior

I expected to get a position near where i am.

### Reproduction steps

I guess you could try to see if `getLastKnownPosition` works on iOS.

### Configuration

**Version:** 1.x

^8.0.4

**Platform:** 
- [X ] :iphone: iOS
- [ ] :robot: Android

**Comment by erf** [comment]
Hi!

I'm on an real device iPhone SE (2 gen) on OS 15.2.1

Thanks for your help!

As a side note, i also experience getting the current position is slow (12sec), even when using `reduced` accuracy.

**Comment by mvanbeusekom** [comment]
Unfortunately the `getCurrentPosition` method on iOS is know to be slow. This is caused be the way the iOS native [requestLocation](https://developer.apple.com/documentation/corelocation/cllocationmanager/1620548-requestlocation) method works. 

Regarding the strange location, do you also get this on other devices? Do you also experience this with other apps on the same device (e.g. Google Maps)? I've experienced similar issues which were caused by a faulty chip.

**Comment by erf** [comment]
Still seem a bit slow.. I did a google search and found this StackOverflow [answer](https://stackoverflow.com/a/42250552/1391965) which says you should call `locationManager.startUpdatingLocation()`, i searched your code repo, and found the native method [requestPosition](https://github.com/Baseflow/flutter-geolocator/blob/2de37b0b891b3c707c0784063ef4ac929803f7e9/geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m#L25) is calling [startUpdatingLocationWithDesiredAccuracy](https://github.com/Baseflow/flutter-geolocator/blob/2de37b0b891b3c707c0784063ef4ac929803f7e9/geolocator_apple/apple/Classes/Handlers/GeolocationHandler.m#L35) and always passing `kCLLocationAccuracyBest`, is this correct or maybe that be why it's slow?

Regarding the strange location on `getLastKnownPosition` i've only tried and experienced that on my iPhone device so far.

**Comment by mvanbeusekom** [comment]
@erf if you have another look at the `requestPosition` method you will see that on iOS 9.0 and higher it will in fact call the `requestLocation` method and return (never reaching the `startUpdatingLocationWithDesiredAccuracy` call):

```objc
- (void)requestPosition:(GeolocatorResult _Nonnull)resultHandler
           errorHandler:(GeolocatorError _Nonnull)errorHandler {
  self.errorHandler = errorHandler;
  self.resultHandler = resultHandler;
  
  if (@available(iOS 9.0, macOS 10.14, *)) {
    [self.locationManager requestLocation];
    return;
  }
  
  [self startUpdatingLocationWithDesiredAccuracy:kCLLocationAccuracyBest
                                  distanceFilter:kCLDistanceFilterNone
               pauseLocationUpdatesAutomatically:NO
                                    activityType:CLActivityTypeOther];
}
```
Unfortunately this method can be slow and is explained more detailed in this [comment](https://github.com/Baseflow/flutter-geolocator/issues/887#issuecomment-958123890). As explained in that comment you can listen to the `getPositionStream()` method which under the hood uses `locationManager.startUpdatingLocation()`, this however has the problem that the first result is the last known location (which often is not very precise)  and the second result might be a less accurate location as well.

Back to the original problem, can you verify if other apps also experience location problems on this device? I strongly suspect there is a hardware problem.

**Comment by erf** [comment]
Ok, maybe i should give `getPositionStream()` a try, and just skip the initial event.

Not sure how to test that. Don't think i've seen issues like that before. I could try to find a Android device to test on it.

**Comment by erf** [comment]
I don't get the weird location when using `getPositionStream()`s first and it seem to work much better. Thanks!

**Comment by BaechlerM-Liip** [comment]
Don't want to create another issue has this is a question related to one comment in here.

> @erf if you have another look at the `requestPosition` method you will see that on iOS 9.0 and higher it will in fact call the `requestLocation` method and return (never reaching the `startUpdatingLocationWithDesiredAccuracy` call):
> 
> ```objc
> - (void)requestPosition:(GeolocatorResult _Nonnull)resultHandler
>            errorHandler:(GeolocatorError _Nonnull)errorHandler {
>   self.errorHandler = errorHandler;
>   self.resultHandler = resultHandler;
>   
>   if (@available(iOS 9.0, macOS 10.14, *)) {
>     [self.locationManager requestLocation];
>     return;
>   }
>   
>   [self startUpdatingLocationWithDesiredAccuracy:kCLLocationAccuracyBest
>                                   distanceFilter:kCLDistanceFilterNone
>                pauseLocationUpdatesAutomatically:NO
>                                     activityType:CLActivityTypeOther];
> }
> ```
> 
> Unfortunately this method can be slow and is explained more detailed in this [comment](https://github.com/Baseflow/flutter-geolocator/issues/887#issuecomment-958123890). As explained in that comment you can listen to the `getPositionStream()` method which under the hood uses `locationManager.startUpdatingLocation()`, this however has the problem that the first result is the last known location (which often is not very precise) and the second result might be a less accurate location as well.
> 
> Back to the original problem, can you verify if other apps also experience location problems on this device? I strongly suspect there is a hardware problem.

So, for iOS 9 and more, the `desiredAccuracy` specified in `getCurrentPosition()` is never used? Is it really the behavior wanted?
